### PR TITLE
Onboarding Jenkins prod docker images to github ci check

### LINKED ISSUE #1455: [Campaign] Ensure Github workflow runs on docker image used by Production Distribution Build
Hi All, 

This is coming from the campaign here: 
* https://github.com/opensearch-project/opensearch-build/issues/3966

### Overview
We would like your CI check (specifically plugin build) in GitHub Repo to run on top of the Build Docker Images from production distribution pipeline.

This is to ensure every plugin repo will use the exact docker images we used in Jenkins build, to check their PRs and run tests before merging the code, so that issues can be detected earlier, and environment can be identical across teams.

### Solutions
The Build Team has created a simple script to dynamically retrieve the current docker image name/tag, so everyone can easily pull the images for their CI checks.

We have a trial run of the above with k-NN team. The script retrieves the docker image dynamically, save output, and use it as the docker image to pull for the upcoming run:
* https://github.com/opensearch-project/k-NN/pull/1099

Note that GitHub Actions only support LINUX docker container at the time of this writing, so we will add Windows containers later on as well as macOS.

### Implementation Notes
We would like you to review above PR and implement similar changes. Note on line `33` of the above k-NN PR, `-u` and `-p` parameters needs to assign values accordingly. 

* OpenSearch Plugin:
```
          CI_IMAGE_VERSION=`opensearch-build/docker/ci/get-ci-images.sh -p centos7 -u opensearch -t build | head -1`
```

* OpenSearch-Dashboards Plugin:
```
          CI_IMAGE_VERSION=`opensearch-build/docker/ci/get-ci-images.sh -p rockylinux8 -u opensearch-dashboards -t build | head -1`
```

**Note that in the above k-NN PR, despite it being OpenSearch plugin, it still uses `rockylinux8`, as we initially plan to upgrade to rockylinux. We have since revert back to `centos7` to support older versions of systems running k-NN lib. As a result, all OpenSearch plugins still uses `centos7` for the time being, and all OpenSearch-Dashboards plugins can go to `rockylinux8`.**

### Completion Date
The above should be implemented by `Nov. 1, 2023 (2023-11-01)` by Plugin Owners to their repository.
And backport the changes to `2.x` branch after merging in `main` branch.

### Contacts
Please contact @peterzhuamazon for any questions on this campaign.

cc: @bbarani

Thanks.

### PR DESCRIPTION
**Pull request by peterzhuamazon**
### Description
Onboarding Jenkins prod docker images to github ci check
 
### Issues Resolved
#1455
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by ylwu-amzn — APPROVED** [review]

File: .github/workflows/CI-workflow.yml [hunk 1]
```diff
@@ -41,43 +55,86 @@ jobs:
...
-        run: |
-          ./gradlew publishToMavenLocal
-      - name: Multi Nodes Integration Testing
-        run: |
-          export OPENAI_KEY=$(aws secretsmanager get-secret-value --secret-id github_openai_key --query SecretString --output text)
-          export COHERE_KEY=$(aws secretsmanager get-secret-value --secret-id github_cohere_key --query SecretString --output text)
-          echo "::add-mask::$OPENAI_KEY"
-          echo "::add-mask::$COHERE_KEY"
-          ./gradlew integTest -PnumNodes=3
+          chown -R 1000:1000 `pwd`
```
Comment (dhrubo-os) [inline .github/workflows/CI-workflow.yml:60]: Just from curiosity: is user 1000 and group id is also 1000? And where this user is defined?

File: .github/workflows/CI-workflow.yml [hunk 2]
```diff
@@ -41,43 +55,86 @@ jobs:
...
+          su `id -un 1000` -c 'whoami && java -version && 
+                               export OPENAI_KEY=`aws secretsmanager get-secret-value --secret-id github_openai_key --query SecretString --output text` &&
+                               export COHERE_KEY=`aws secretsmanager get-secret-value --secret-id github_cohere_key --query SecretString --output text` &&
+                               echo "::add-mask::$OPENAI_KEY" &&
+                               echo "::add-mask::$COHERE_KEY" &&
+                               ./gradlew build &&
+                               ./gradlew publishToMavenLoca &&
```
Comment (dhrubo-os) [inline .github/workflows/CI-workflow.yml:67]: publishToMavenLoca --> `publishToMavenLocal`

File: .github/workflows/CI-workflow.yml [hunk 3]
```diff
@@ -41,43 +55,86 @@ jobs:
...
+                               ./gradlew integTest -PnumNodes=3'
```
Comment (dhrubo-os) [inline .github/workflows/CI-workflow.yml:68]: Can we add comment? Previously we had step names like: `Multi Nodes Integration Testing` which helps to understand. May be we can add these step names as a comment for better clarification?

File: .github/workflows/CI-workflow.yml [hunk 4]
```diff
@@ -41,43 +55,86 @@ jobs:
...
+          plugin=`basename $(ls plugin/build/distributions/*.zip)`
+          echo $plugin
+          mv plugin/build/distributions/$plugin ./
```
Comment (dhrubo-os) [inline .github/workflows/CI-workflow.yml:71]: Curious to know the output of `plugin/build/distributions/$plugin`

File: .github/workflows/CI-workflow.yml (hunk 3 above)
Comment (peterzhuamazon) [reply .github/workflows/CI-workflow.yml:68]: Sure.

File: .github/workflows/CI-workflow.yml (hunk 2 above)
Comment (peterzhuamazon) [reply .github/workflows/CI-workflow.yml:67]: Good catch!

File: .github/workflows/CI-workflow.yml (hunk 4 above)
Comment (peterzhuamazon) [reply .github/workflows/CI-workflow.yml:71]: I will change it to `mv -v` so you will see the entire output.

**Review by dhrubo-os — APPROVED** [review]

**Review by jngz-es — APPROVED** [review]

**Pull request merged**